src/utils/dom.ts

```diff
@@ -6,20 +6,26 @@
 
 import * as d3 from "d3";
 
-export type AnySelection = d3.Selection<any, any, any, any>;
+export type d3Selection<D> = d3.Selection<any, D, any, any>;
 
 export class DOM {
-  public static transform(s: AnySelection, x: number, y: number): AnySelection {
+  public static transform(s: d3Selection<any>): string;
+  public static transform(s: d3Selection<any>, x: number, y: number): d3Selection<any>;
+  public static transform(s: d3Selection<any>, x?: number, y?: number): any {
+    if (x == null) {
+      return s.attr("transform");
+    } else {
       y = (y == null) ? 0 : y;
       const translate = `translate(${x}, ${y})`;
       s.attr("transform", translate);
       return s;
+    }
   }
 
-  public static getBBox(element: AnySelection): SVGRect {
+  public static getBBox(element: d3Selection<any>): SVGRect {
```

Comment:
`d3.Selection` accepts `BaseType`, so common parent class with `SVGLocatable ` would be `any`. Also we wrap it with try/catch, so we can accept anything and return empty box in case of missing method.